### PR TITLE
TM-87 Adding My Pledges page

### PR DESCRIPTION
**Pull request by ConorTill**
Dependent on TM-24 and TM-43

### CODE REVIEW

**Comment by DannyGB** [comment]
Can we also get the SonarCloud issues back down to the minimum please?

File: src/SFA.DAS.LevyTransferMatching.Web/Models/Opportunities/IndexViewModel.cs [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+        {
+            public int Amount { get; set; }
+            public string EmployerName { get; set; }
+            public string ReferenceNumber { get; set; }
+            public IEnumerable<string> Locations { get; set; }
+            public string Sectors { get; set; }
+            public string JobRoles { get; set; }
+            public string Levels { get; set; }
+            public string DisplayAmount => Amount.ToString("C0", new CultureInfo("en-GB"));
+            public string DisplayLocations => (Locations != null && Locations.Any()) ? string.Join(", ", Locations) : "All";
```
Comment (DannyGB) [inline src/SFA.DAS.LevyTransferMatching.Web/Models/Opportunities/IndexViewModel.cs:24]: You could use the IsComplete() extension here to replace
`(Locations != null && Locations.Any())`

File: src/SFA.DAS.LevyTransferMatching.Web/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -28,5 +29,15 @@ public static string ToReferenceDataDescriptionList(this IEnumerable<string> sel
...
+
+        public static bool IsComplete(this string str)
+        {
+            return !string.IsNullOrEmpty(str) && str != NotEntered;
+        }
+
+        public static bool IsComplete(this IEnumerable<string> list)
+        {
+            return list != null && list.Any();
+        }
```
[the remark this answers is not in the document]
Comment (ConorTill) [reply src/SFA.DAS.LevyTransferMatching.Web/Extensions/StringExtensions.cs]: As far as I can tell both of these methods are in main labelling you as the author? Not sure what's happened here really, maybe a merge error snuck through in a previous ticket?

**Review by mmiddleton3301 — APPROVED** [review]
LGTM ✅ Can't pick up on anything here that needs changing.

**Pull request merged**